registries/docker-daemon/configure.sh

```diff
@@ -3,15 +3,21 @@
 daemonConfig='/etc/docker/daemon.json'
 if test -f ${daemonConfig} && grep -q registry.kube-system.svc.cluster.local ${daemonConfig}; then
   echo "insecure registry previously configured"
-else
-  # Allow for insecure registries
-  sudo mkdir -p /etc/docker
-  echo '{ "insecure-registries": [ "registry.kube-system.svc.cluster.local:5000" ] }' | sudo tee ${daemonConfig} > /dev/null
-  sudo systemctl daemon-reload
-  sudo systemctl restart docker
+elif which systemctl > /dev/null; then
+  if systemctl status docker 2>&1 > /dev/null; then
+    # Allow for insecure registries as long as docker daemon is actually running
+    if ! test -f ${daemonConfig} || test -s ${daemonConfig}; then
+      sudo mkdir -p /etc/docker
+      echo '{}' | sudo tee ${daemonConfig} > /dev/null
+    fi
+    jq -s '.[0] * .[1]' <( cat ${daemonConfig}) <(echo '{ "insecure-registries": [ "registry.kube-system.svc.cluster.local:5000" ] }') | sudo tee ${daemonConfig} > /dev/null
+    sudo systemctl daemon-reload
+    sudo systemctl restart docker
+  fi
 fi
 
 IMAGE_REPOSITORY_PREFIX="registry.kube-system.svc.cluster.local:5000"
+CLUSTER_NAME=${CLUSTER_NAME-fats}
```

Comment:
rm

CLUSTER_NAME is part of the FATS contract. We should not define it here. The script should fail if the value is not set.